src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -705,13 +705,13 @@ protected override void Dispose(bool disposing)
                     }
                     finally
                     {
-                        _deflater = null;
-                        _inflater = null;
+                        _deflater = null!;
```

Comment:
This `!` shouldn't be needed anymore, you made `_deflater`, `_inflater` and `_buffer` nullable already.